source/fundamentals/typescript.txt

```diff
@@ -142,6 +142,75 @@ section.
 
 .. _node-driver-limitations:
 
+Find Methods and the _id Field
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``find`` and ``findOne`` methods always include the ``_id`` field in their
+return type.
+
+If the type parameter you passed to your ``Collection`` instance includes the
+``_id`` field, the type of the ``_id`` field in the values returned from the 
+find methods is the same as the type of the ``_id``
+field in your type parameter.
+
+If the type parameter passed to your ``Collection`` instance does not include the ``_id``
+field, the driver gives the ``_id`` field in the values returned from the find methods
+the type ``ObjectId``.
+
+The following code uses the :ref:`Pet <mongodb-node-typescript-pet-interface>` interface to return
+a document with an ``_id`` of type ``ObjectId``:
+
+.. code-block:: typescript
+
+   const database = client.db("<your database>");
+   const collection = db.collection<Pet>("<your collection>");
+   
+   const document = await collection.findOne({
+     name: "Spot",
+   });
+   const id : ObjectId = document._id;
+
+The following code uses the ``IdNumberPet`` interface to return a
+document with an ``_id`` field of type ``number``:
+
+.. code-block:: typescript
+
+   interface IdNumberPet {
+     _id: number;
+     name: string;
+     age: number;
+   }
+
+   const database = client.db("<your database>");
+   const collection = db.collection<IdNumberPet>("<your collection>");
+
+   const document = await collection.findOne({
+     name: "Spot",
+   });
+   const id : number = document._id;
+
+.. important:: Projection
+
+   If you specify a projection that removes the ``_id`` field in a find method,
+   you must manually ensure that you do not reference the ``_id`` field of values
+   returned by that method. TypeScript cannot check at compile time that a referenced
+   ``_id`` field was excluded with a projection.
```

Comment:
it may be worth mentioning here that we specifically allow overriding the expected type of the find for the use case of projections; if you know what your projection does, you can specify the return type on the find in your code and it will the properly warn you at compile type if you try to access a property that does not exist (see `find<T>`, `findOne<T>` overrides), in this case, you could do `await collection.findOne<PetWithoutId>({}, { projection: { _id: 0 } })` where `PetWithoutId` is the `Pet` type that omits the `_id` (or whatever type the collection uses)